week-3/Homework/extra/1-card-validator.js

```diff
@@ -0,0 +1,82 @@
+// ## **PROJECT: Credit Card Validator**
+
+// In this project you'll write a script that validates whether or not a credit card number is valid.
+
+// Here are the rules for a valid number:
+
+// - Number must be 16 digits, all of them must be numbers
+// - You must have at least two different digits represented (all of the digits cannot be the same)
+// - The final digit must be even
+// - The sum of all the digits must be greater than 16
+// - The following credit card numbers are valid:
+
+// ```markdown
+// 9999777788880000
+// 6666666666661666
+// ```
+
+// The following credit card numbers are invalid:
+
+// ```markdown
+// a92332119c011112 (invalid characters)
+// 4444444444444444 (only one type of number)
+// 1111111111111110 (sum less than 16)
+// 6666666666666661 (odd final number)
+// ```
+
+// These are the requirements your project needs to fulfill:
+
+// - Make a JavaScript file with a name that describes its contents
+// - Create a function with a descriptive name, for example: `doSomething` or `calcAnotherThing`
+// - Write at least 2 comments that explain to others what a line of code is meant to do
+// - Make the return value of the function a template string, so you can insert variables!
+// - Use `node` from the command line to test if your code works as expected
+
+// Good luck!
+
+function validateCreditCard(cardNum) {
+  // Changing the number to an array of single digit strings.
+  let digitsAsStringsArr = String(cardNum).split("");
+  let len = digitsAsStringsArr.length;
+  let isNumber = /[0-9]/;
+  // If any of the following conditions is true the number will be invalid.
+  if (
+    len !== 16 ||
+    digitsAsStringsArr.some(element => !isNumber.test(parseFloat(element))) ||
```

Comment:
Regular expressions are things we test strings against - here you're converting the string to a number, and then testing it against the regular expression. You don't need this `parseFloat `here - that said, javascript will convert that float back to a string to do the test for you, but it's extra work that could be avoided :)

Reply:
Ohh, that is very good to know. I deleted the unnecessary parseFloat.